src/zenml/cli/stack.py

```diff
@@ -1250,3 +1278,316 @@ def register_secrets(
         cli_utils.declare(f"Updating secret `{secret.name}`:")
         cli_utils.pretty_print_secret(secret=secret, hide_secret=True)
         secrets_manager.update_secret(secret)
+
+
+@stack.command(help="Deploy a stack using mlstacks.")
+@click.option(
+    "--provider",
+    "-p",
+    "provider",
+    required=True,
+    type=click.Choice(STACK_RECIPE_MODULAR_RECIPES),
+)
+@click.option(
+    "--stack-name",
+    "-n",
+    "stack_name",
+    type=click.STRING,
+    required=True,
+    help="Set a name for the ZenML stack that will be imported from the YAML "
+    "configuration file which gets generated after deploying the stack recipe. "
+    "Defaults to the name of the stack recipe being deployed.",
+)
+@click.option(
+    "--region",
+    "-r",
+    "region",
+    type=click.STRING,
+    required=True,
+    help="The region to deploy the stack to.",
+)
+@click.option(
+    "--import",
+    "-i",
+    "no_import_stack_flag",
+    is_flag=True,
+    help="If you don't want the stack to be imported automatically.",
+)
+@click.option(
+    "--artifact-store",
+    "-a",
+    "artifact_store",
+    required=False,
+    is_flag=True,
+    help="Whether to deploy an artifact store.",
+)
+@click.option(
+    "--container-registry",
+    "-c",
+    "container_registry",
+    required=False,
+    is_flag=True,
+    help="Whether to deploy a container registry.",
+)
+@click.option(
+    "--mlops-platform",
+    "-m",
+    "mlops_platform",
+    type=click.Choice(["zenml"]),
+    required=False,
+    help="The flavor of MLOps platform to use."
+    "If not specified, the default MLOps platform will be used.",
+)
```

Comment:
Do you mean why are we allowing people to deploy ZenML using `mlstacks`? We
already offer it in our CLI, and potentially we might have other platforms we
could support with the recipes?

I guess I'd ask what the reason is for your hesitation to add this here?

Reply:
We offer it because someone might not always want to deploy ZenML. For instance,
for someone who wants to just deploy mlflow and an artifact store, we shouldn't
force them also to deploy ZenML.